packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -971,6 +979,13 @@ export type IDirectiveResolvers${contextType} = ${name}<ContextType>;`
       });
       const mappedTypeKey = isSubscriptionType ? `${mappedType}, "${node.name}"` : mappedType;
 
+      const directiveMappings =
+        node.directives
+          ?.map(directive => this._customDirectiveToResolverTypeNameMappings.get(directive.name as any))
+          .filter(Boolean) ?? [];
+
+      const resolverType = isSubscriptionType ? 'SubscriptionResolver' : directiveMappings[0] ?? 'Resolver';
```

Comment:
We take the first, I don't think this should be over-used. My ideal use-case is having ``@unauthenticated` for only marking those fields that should be available with auth and thus making the viewer within that resolver optional 🤔